api/envoy/config/route/v3/route_components.proto

```diff
@@ -1629,6 +1629,24 @@ message RateLimit {
       string default_value = 3;
     }
 
+    // The following descriptor entry is appended when the route entry metadata contains a key value:
+    //
+    // .. code-block:: cpp
+    //
+    //   ("<descriptor_key>", "<value_queried_from_route_entry_metadata>")
+    message RouteEntryMetaData {
+      // The key to use in the descriptor entry.
+      string descriptor_key = 1 [(validate.rules).string = {min_len: 1}];
+
+      // Metadata struct that defines the key and path to retrieve the string value. A match will
+      // only happen if the value in the filter metadata is of type string.
```

Comment:
filter metadata or route entry metadata?

Reply:
route entry metadata. This is slipped through the renaming.